src/AsyncGenerator.yml

```diff
@@ -139,6 +142,10 @@
     - conversion: Ignore
       name: DependentContext
       containingTypeName: AdoNetWithSystemTransactionFactory
+    asyncExtensionMethods:
+      projectFiles:
+      - fileName: AsyncExtensions.cs
```

Comment:
Am I doing something wrong, @maca88?

Reply:
Upgrade to AsyncGenerator version `0.21.0` as it include some fixes regarding extension methods. As you will find out, the generated code does not compile for `.netstandard2.1` as `BeginTransactionAsync` return `ValueTask` and not `Task`. Therefore I suggest to first merge #3000 and then update the `targetFramework` of project `NHibernate` from `netcoreapp2.0` to `net6.0`.

Reply:
I'll try to make it working on .net core 2.0 (bugging you by the way, so sorry, but please bear with me)